libraries/featureflag/impl/src/main/kotlin/io/element/android/libraries/featureflag/impl/StaticFeatureFlagProvider.kt

```diff
@@ -20,19 +20,18 @@ import io.element.android.libraries.featureflag.api.Feature
 import io.element.android.libraries.featureflag.api.FeatureFlags
 import javax.inject.Inject
 
-class BuildtimeFeatureFlagProvider @Inject constructor() :
+/**
+ * This provider is used for release build.
+ * [isFeatureEnabled] just returns the default value of the FeatureFlags.
+ */
+class StaticFeatureFlagProvider @Inject constructor() :
     FeatureFlagProvider {
 
-    override val priority: Int
-        get() = LOW_PRIORITY
+    override val priority = LOW_PRIORITY
 
     override suspend fun isFeatureEnabled(feature: Feature): Boolean {
         return if (feature is FeatureFlags) {
-            when (feature) {
-                FeatureFlags.LocationSharing -> true
-                FeatureFlags.Polls -> false
-                FeatureFlags.NotificationSettings -> false
-            }
+            feature.defaultValue
```

Comment:
IMO StaticFeatureFlag should really be clear about which feature is enabled or not.
Using a `when` makes it exhaustive, so it's the responsability of the developer to enable it or not before the release. Maybe we should just add a comment at the beginning of the when to make it clear

Reply:
Developers forget to update the value here, they just do it in `FeatureFlags`, like for instance in this PR: https://github.com/vector-im/element-x-android/pull/1196.

With this change, I think there is no more possible mistake 🤞

Reply:
We should just educate developers, it should be clear which feature or not is enabled in release...

Reply:
I see. I will try to improve this then.

Ideally we would have a centralized configuration file, that can also be used for gradle Kotlin files.

Reply:
Yes but we should makes sure it's exhaustive...